Jint/Runtime/Interop/ObjectWrapper.cs

```diff
@@ -166,7 +136,9 @@ private IEnumerable<JsValue> EnumerateOwnPropertyKeys(Types types)
                 // we take values exposed as dictionary keys only
                 foreach (var key in dictionary.Keys)
                 {
-                    if (_engine.ClrTypeConverter.TryConvert(key, typeof(string), CultureInfo.InvariantCulture, out var stringKey))
+                    object stringKey = key as string;
+                    if (stringKey is not null
+                        || _engine.ClrTypeConverter.TryConvert(key, typeof(string), CultureInfo.InvariantCulture, out stringKey))
```

Comment:
Would probably work without the custom IObjectConverter by checking for `IConvertible` and ultimately for `IFormattable` if we know we want a string.

Reply:
That's how Fluid can handle `JToken` without configuration.

Reply:
Here I just wanted to create a fast path when we already have a string, but might be worth some PR to improve string coercion